### PR TITLE
Upgrade dummy app and compatibility to rails 7.2

### PR DESCRIPTION
**Pull request by sfnelson**

### CODE REVIEW

File: spec/dummy/config/puma.rb [hunk 1]
```diff
@@ -1,10 +1,12 @@
+# frozen_string_literal: true
+
 # Puma can serve each request in a thread from an internal thread pool.
 # The `threads` method setting takes two numbers: a minimum and maximum.
 # Any libraries that use thread pools should be configured to match
 # the maximum value specified for Puma. Default is set to 5 threads for minimum
 # and maximum; this matches the default thread size of Active Record.
 #
-max_threads_count = ENV.fetch("RAILS_MAX_THREADS") { 5 }
+max_threads_count = ENV.fetch("RAILS_MAX_THREADS", 5)
```
Comment (AlanCornthwaiteKatalyst) [inline spec/dummy/config/puma.rb:9]: Hasn't the recommendation from rails changed from 5 to 3?

File: app/controllers/katalyst/content/items_controller.rb [hunk 2]
```diff
@@ -87,15 +87,32 @@ def render_editor(**)
       #
       # This mimics the behaviour of direct uploads without requiring the JS
       # integration.
+      #
+      # Note: this idea comes from various blobs who have documented this approach, such as
```
Comment (AlanCornthwaiteKatalyst) [inline app/controllers/katalyst/content/items_controller.rb:91]: typo in note, various `blogs`

**Review by AlanCornthwaiteKatalyst — COMMENTED** [review]
Looks good to me, a small typo and 1 question re puma thread counts.

**Review by AlanCornthwaiteKatalyst — APPROVED** [review]

**Review by hasarindaKI — APPROVED** [review]

File: spec/dummy/config/puma.rb (hunk 1 above)
Comment (sfnelson) [reply spec/dummy/config/puma.rb:9]: It's irrelevant, we're never going to run a dummy app. I just let the templates do their thing.

File: app/models/concerns/katalyst/content/container.rb [hunk 3]
```diff
@@ -49,8 +49,24 @@ def inactive
...
+          %w[published draft].include?(state)
+        end
+
+        def draft?
+          state == "draft"
+        end
+
+        def unpublished?
+          state == "unpublished"
+        end
```
Comment (sobakasu) [inline app/models/concerns/katalyst/content/container.rb:69]: (opt) something that's been very popular of late in gus:

```
delegate :draft, :unpublished, to: :state

def state
    (super || "").inquiry
end

def published? = state.published? || state.draft?
```

Comment (sfnelson) [reply app/models/concerns/katalyst/content/container.rb:69]: I don't really see the benefit. Saves characters? It's a bit hard to understand imo.

**Pull request merged**